plugins/module_utils/network/vyos/config/ospf_interfaces/ospf_interfaces.py

```diff
@@ -61,12 +65,36 @@ def __init__(self, module):
             "passive",
         ]
 
+    def _validate_template(self):
+        if self._module.params.get("version") == "detect":
+            try:
+                version = self._module._connection.get_device_info()["network_os_major_version"]
+            except (KeyError, AttributeError) as e:
+                version = "1.2"  # default to 1.2 if no connection
+        else:
+            version = self._module.params.get("version")
+        if version >= "1.4":
```

Comment:
Thanks, @omnom62. Will definitely fix the comparison.  The main reason for the location here was making sure we could override during unit tests. I will admit I'm a long-time Ansible user and Python programmer, however my experience with the network modules is limited, so part of my concern is making sure we put this in the right place.

Happy to look at moving this if we can find the right location that makes it more efficient and can be counted upon anywhere (and preferably something that can be overridden by the user in case of operations that are being done in the configuration in the absence of a connection).

Reply:
All good mate. The thing should work as is for now - given the priority to get the merge out of the door sooner, we can schedule some performance review or other refactor later or when it becomes too evident ;)

Reply:
Looking around, I think the refactor (and likely use of packaging.version) may be appropriate course after we merge this. There are a number of places where I needed to check for versions and they're mostly similar, but none of them are safe for >1.9.